src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -451,7 +451,11 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let newConnectionProfile = new ConnectionProfile(this.notebookOptions.capabilitiesService, newConnection);
 			this._activeConnection = newConnectionProfile;
 			this.refreshConnections(newConnectionProfile);
-			await this._activeClientSession.updateConnection(this._activeConnection);
+			this._activeClientSession.updateConnection(this._activeConnection.toIConnectionProfile()).catch((error) => {
+				if (error) {
+					this.notifyError(error.message);
```

Comment:
Minor: we have a util method to get the error message. I think it's `notebookUtils.getErrorMessage`. This handles cases where it's a string error in addition to Error class